lib/recipe/widgets/ingredient_form.dart

```diff
@@ -1,14 +1,31 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:dropdown_search/dropdown_search.dart';
+import 'package:flutter_slidable/flutter_slidable.dart';
+import 'dart:convert';
+
+import 'package:gibsonify_api/gibsonify_api.dart';
 import 'package:gibsonify/recipe/recipe.dart';
-import 'package:dropdown_plus/dropdown_plus.dart';
 
-class IngredientForm extends StatelessWidget {
+class IngredientForm extends StatefulWidget {
   final int recipeIndex;
   final int ingredientIndex;
-  IngredientForm(this.recipeIndex, this.ingredientIndex, {Key? key})
+  const IngredientForm(this.recipeIndex, this.ingredientIndex, {Key? key})
       : super(key: key);
 
+  @override
+  State<IngredientForm> createState() => _IngredientFormState();
+}
+
+class _IngredientFormState extends State<IngredientForm> {
+  Future<String>? _ingredients;
+
+  @override
+  void initState() {
+    super.initState();
+    _ingredients = Ingredient.getIngredients();
+  }
+
```

Comment:
I think it would be more suitable from a BLoC point of view to load the ingredients to the state of the BLoC, maybe when initializing the BLoC. The UI (the form) should ideally be Stateless and the BloC state would have the ingredients.

Reply:
I think there's a bit of a conflict of best design practice here between native Flutter and bloc - from what I gathered it was best practice to only load an asset when we need it (when we open the ingredient form), which is why I had the logic there. In a world where we may have more (or larger) assets, it may become a bigger issue if we choose to preload assets in the state. That said, I can see the elegance in loading it right into the state and am happy to do so if it makes sense, perhaps in a future PR depending on if I can get it done before the handover.

Reply:
Yeah, let's just make a `TODO` now and refactor it later, I'd like this to be merged soon so that I can reuse the `Measurement` class.

Reply:
Sounds good to me, will add the TODO.